### ISSUE TITLE
Toolstack 'xm' has been deprecated in favor of 'xl'

### ISSUE THREAD

**Issue by GhostLyrics** [sensu-plugins/sensu-plugins-xen#2]
`xm` is [deprecated](http://wiki.xen.org/wiki/XL). In theory one should be able to simply exchange all occurrences of `xm` in shell commands with `xl` and you should be good to go.

For an upcoming PR: Is it preferable to keep the script simple and provide 2 versions:
- bin/metrics-xen-xm
- bin/metrics-xen-xl

OR rewrite the script with some detection
OR rewrite the script and don't be backwards compatible?

**Comment by eheydrick** [comment]
Let's replace `xm` with `xl` in the same script and release a new major version. xm has been deprecated since 2012 so plenty of time for users to upgrade to xl.

**Comment by mag009** [comment]
We use Debian and apparently Debian by default use xm as toolstack... Not sure if it's still the case with Jessie ( I hope they changed it to xl...).

I'll try to grab some time on Friday to support xl.

**Comment by GhostLyrics** [comment]
@mag009 Have you had the time to do some work here? I also considered taking a stab. In addition I considered adding a check to make sure a VM is running.

e.g. if you were to use that on 4 VM hosts, you'd know that at least one is currently running that mission-critical VM you need.

**Comment by mag009** [comment]
sorry bit late on that one ;) busy at work I spent some time to finally do it. I've added this quickly via command line so you can chose which one you wanna use. As mentioned, I still use the xm command but since Debian 8 we now needed xl so. I only use xm|xl to get static value like the nr cores ram etc.. eventually I'll grab those from /proc.. instead of depending on tool like that.

let me know if there's any problem.

cheers

**Comment by GhostLyrics** [comment]
I'll test the new version by checking out your PR onto a test machine.

I'm not sure what you want to get from `/proc/xen`. On my production machine there are `capabilities`, `xenbus`, `privcmd`, `xsd_kva` and `xsd_ports`. When I `cat` them:

(capabilities):

```
control_d
```

(xenbus):

```
(nothing, command does not return)
```

(privcmd):

```
cat: privcmd: invalid argument
```

(xsd_kva):

```
0xffff8800593ff000
```

(xsd_port):

```
113
```

**Comment by mag009** [comment]
not /proc/xen but system like /proc/cpuinfo, meminfo etc.. xl|xm is only used to output the nr_cores that I have on my baremetal I use those as limit in my graph's to get an overview of how much I over commit my vcpu etc..

**Comment by majormoses** [comment]
@mag009 this can be closed now that this: #3  has landed and released right?

**Comment by mag009** [comment]
yep case closed thanks @majormoses
